lib/shoulda/matchers/active_model/validate_numericality_of_matcher.rb

```diff
@@ -377,6 +377,8 @@ def initialize(attribute)
           @allowed_type_name = 'number'
           @context = nil
           @expected_message = nil
+
+          add_disallow_non_numeric_value_matcher
```

Comment:
I changed that behavior to the initializer because by doing this, we can ensure that the first matcher to be checked is always the one to disallow non-numerical values. With that, we can display more intuitive failure messages.